app/models/parser/twitter_profile.rb

```diff
@@ -9,10 +9,7 @@ def type
 
       def patterns
         [
-          /^https?:\/\/(www\.)?twitter\.com\/(?<username>[\w\d]+)(\?+.*)$/,
-          /^https?:\/\/(0|m|mobile)\.twitter\.com\/(?<username>[\w\d]+)(\?+.*)$/,
-          /^https?:\/\/(www\.)?twitter\.com\/(?<username>[^\/]+)$/,
-          /^https?:\/\/(0|m|mobile)\.twitter\.com\/(?<username>[^\/]+)$/
+          /^https?:\/\/((0|m|mobile|www)\.)?twitter\.com\/(?<username>[\w]{4,15})(\?.*)?[\/]*$/
```

Comment:
Good job! Minor change, but needed: `/` should come before the query parameters, so it can match something like this: `https://twitter.com/foo_bar/?t=1`.